src/IteratorAwareTrait.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Dhii\Collection;
+
+use Iterator;
+use Traversable;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+use OutOfRangeException;
+use Exception as RootException;
+
+/**
+ * Awareness of an iterator.
+ *
+ * @since [*next-version*]
+ */
+trait IteratorAwareTrait
```

Comment:
We already have an `IteratorAwareTrait`, albeit in a different namespace. To avoid confusion, I recommend renaming this trait, especially since it's mostly a trait for awareness of a _data store_ iterator. Suggestion: `DataStoreIteratorTrait`.

Reply:
It's not the same iterator, and that's clear because it's in a different namespace - it's an iterator-aware trait used in collections.

Reply:
Up to you. I just think the name doesn't do the implementation justice.

Reply:
I'll think about it a bit more tomorrow.